R/reportDIETER.R

```diff
@@ -0,0 +1,284 @@
+#' Reporting for the coupled DIETER Model
+#'
+#'
+#' *Warning* The function modifies the "REMIND_generic_<scenario>.mif" file by appending the
+#' additional reporting variables.
+#'
+#' 
+#' @param dieterDatafile full path with name of dieter gdx file.
+#' @param outputDir path to the output folder, default is current folder.
+#' @author Chen Gong Pratik Agrawal
+#'
+
+#' @importFrom data.table fread fwrite setDT
+#' @importFrom quitte read.quitte read.gdx revalue.levels
+#' @importFrom readr write_rds
+#' @importFrom rlang .data
+#' @importFrom dplyr %>% left_join filter select rename mutate arrange ungroup across mutate_all coalesce
+#' @importFrom tidyr complete
+#' @importFrom stringr str_extract str_trim 
+#' @export
+reportDIETER <- function(dieterDatafile = "report_DIETER.gdx", outputDir=".") {
+  datapath <- function(fname) {
+    file.path(outputDir, fname)
+  }
+  
+  subFolder <- "DIETER"
+  ## NULL Definitons for codeCheck compliance
+  
+  outAnnual <- annual_new <- dieter.tech.mapping <- dieter.tech.mapping.standalone <- NULL
+  annual_remind <- dieter.tech.mapping.REMIND <- remind_miffiles <- out_hourly <- NULL
+  hourly_new <- HourlyandAnnual <- NULL
+  
+  gdxToQuitteHourly <- function(gdxfile) {
+    file <- datapath(fname = gdxfile)
+    out_hourly <- NULL
+    ######################################################################################################################## 
+    rep_hrs <- read.gdx(gdxName = file, requestList.name =  "report_hours", factors = FALSE, squeeze = FALSE) 
+    
+    names(rep_hrs) <- c("gdxfile", "model", "year", "country", "variable", "hour", "value")
+    
+    out_h <- rep_hrs %>% 
+      dplyr::select_(~model, ~year, ~variable, ~country,~hour,~value) %>%
+      dplyr::mutate_(hour = ~as.numeric(stringr::str_extract(hour, "[0-9]+"))) %>% 
+      dplyr::group_by_(~model, ~variable, ~year, ~country) %>%
+      complete(hour = (1:8760)) %>%
+      #replace(is.na(.), 0) %>%
+      dplyr::mutate_all(coalesce, 0)%>%
+      dplyr::ungroup(c('model', 'variable', 'year', 'country')) %>%
+      dplyr::mutate_(Model = ~model, Scenario = ~paste0("baseline"), Region = ~country,
```

Comment:
This paste is not doing anything, is it?
```suggestion
      dplyr::mutate_(Model = ~model, Scenario = ~baseline, Region = ~country,
```